### ISSUE TITLE
add xxd via vim as a dep for ubuntu?

### ISSUE THREAD

**Issue by alecGraves** [PX4/PX4-Devguide#685]
Put ```apt-get install vim``` in ubuntu_sim_nuttx.sh

like with  PR #301 

I was building on ~~docker~~ **a stock ubuntu:16.04 docker image that I set up with  ubuntu_sim_nuttx.sh** and got error that xxd was not installed:

```bash
make px4fmu-v5_default
-- PX4 VERSION: v1.8.0-1490-gf637f6a
# BLAH BLAH BAH ...
-- ROMFS: px4fmu_common
CMake Error at ROMFS/CMakeLists.txt:157 (message):
  xxd not found
```

as the msg implies, xxd is used to do something in ROMFS/CMakeLists.txt

```CMake
# create nsh_romfsimg.c
find_program(XXD xxd)
if(NOT XXD)
    message(FATAL_ERROR "xxd not found")
endif()
add_custom_command(OUTPUT nsh_romfsimg.c
        COMMAND ${CMAKE_COMMAND} -E remove -f nsh_romfsimg.c
        COMMAND ${XXD} -i romfs.img nsh_romfsimg.c
        COMMAND sed 's/unsigned/const unsigned/g' nsh_romfsimg.c > nsh_romfsimg.c.tmp && ${CMAKE_COMMAND} -E rename nsh_romfsimg.c.tmp nsh_romfsimg.c
        DEPENDS romfs.img
        )
```

So after a google search, I find out it is distributed with apt in ```vim-common```.

I installed vim (```apt install vim```) and now it compiles.

Another solution would be to switch to something more common like ```od```?

Anyway, maybe this issue (/ pr if someone makes one) might help someone someday

**Comment by hamishwillee** [comment]
@dagar Docker build error reported here that `xxd` was not installed. @alecGraves fixed this by installing vim, which I would prefer not to do.
1. Can you fix in docker if needed?
2. Does anything need to be done for the devguide scripts? If so, what?

@alecGraves Thanks for reporting this.

**Comment by alecGraves** [comment]
Sorry if there is any confusion, but I am not using the official px4 docker image, just the base ubuntu:16.04 image from dockerhub.
* I installed the deps from the ubuntu convenience scripts "ubuntu_sim_nuttx.sh"
* vim was missing

This will probably not be an issue for most users of ununtu, but if your distro for some reason is missing vim, the firmware build will not work.

**Comment by hamishwillee** [comment]
Thanks @alecGraves . "ubuntu_sim_nuttx.sh" should install everything you need on Ubuntu 16.04, which does not include VIM. If we need VIM then great, but I want @dagar to confirm that this is the best way to get xxd.

@dagar I see you do install vim-common in https://github.com/PX4/containers/blob/master/docker/px4-dev/Dockerfile_nuttx#L22
Why? No other way to get xxd?

**Comment by dagar** [comment]
I believe that's the easiest way on ubuntu.

**Comment by hamishwillee** [comment]
OK, thanks. This should be fixed by #689
